parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala

```diff
@@ -55,11 +56,43 @@ object Tokens extends CrfTokenable {
   override def normalise(tokens: Array[String]): Array[String] = tokens.map(_.toUpperCase)
 
   def tokensToMap(tokens: Seq[CrfTokenResult]): Map[String, String] = {
-    tokens.groupBy(_.label).map {
+    val validatedTokens = validatePostCode(tokens.groupBy(_.label))
```

Comment:
It's a little different to that, this is transforming the tokens to valid map of tokens to be used in the query, changing func name